### ISSUE TITLE
burn 1 nft, mint and transfer nft

### ISSUE THREAD

**Issue by yubrew** [public-awesome/launchpad#338]

**Comment by ivmidable** [comment]
could you clarify what you want on this exactly? it sounds similar to how we did the villain burn/swap for HeroNFT on terra. with the villain burn we allowed them to burn 2 heroes and then it minted them a new villain.

**Comment by ivmidable** [comment]
could you clarify what you want on this exactly? it sounds similar to how we did the villain burn/swap for HeroNFT on terra. with the villain burn we allowed them to burn 2 heroes and then it minted them a new villain.

**Comment by yubrew** [comment]
Yes other creators are asking to do similar things. Some have done it in trusted ways with cosmjs, but I think it would be better to have a smart contract perform the swap

**Comment by ivmidable** [comment]
so here is what I was thinking as far as a v1 for stargaze, instead of burn 1 mint 1 let the collection owner set the burn threshold to mint so it could be burn 2,3,4 to mint 1. 

so at a high level something like new minter contract that uses receiver pattern to get the NFT of a specific collection(maybe multiple collections?) and then it calls burn on it and keeps a count of how many burns each wallet has and when it hits the threshold it will call mint on a new collection(or maybe the same collection?). I think it would open up a lot of interesting use cases.

**Comment by ivmidable** [comment]
could also limit the burning to specific token_ids of a collection that way you could do like we did with Hero and only allow burning of commons or NFTs with specific traits, it would just be a matter of determining a good fairburn fee to charge for setup since it would require storing some extra state data.
